mastodon_archive/core.py

```diff
@@ -221,10 +221,33 @@ def load(file_name, required = False, quiet = False):
         sys.exit(2)
 
     if os.path.isfile(file_name) and os.path.getsize(file_name) > 0:
-        if not quiet:
-            print("Loading existing archive")
-        with open(file_name, mode = 'r', encoding = 'utf-8') as fp:
-            return json.load(fp)
+
+        def _json_load(fname):
+            if not quiet:
+                print("Loading existing archive:", fname)
+
+            with open(fname, mode='r', encoding='utf-8') as fp:
+                return json.load(fp)
+
+        data = _json_load(file_name)
+        if combine:
+            # Load latest archive first to keep chronological order
+            archives = list(
+                reversed(glob.glob(file_name.replace(".json", ".*.json")))
+            )
+
+            if required and not quiet and not archives:
+                print("Warning: No split archives to combine", file=sys.stderr)
+
+            # Merge dictionaries loaded from JSON archives
+            for archive in archives:
+                archived_data = _json_load(archive)
+
+                for collection in ["statuses", "favourites", "mentions"]:
+                    data[collection].extend(archived_data[collection])
```

Comment:
I wonder whether we need to sort afterwards?

Reply:
I wonder what happens to boosts (I should re-check). The `text` sub-command does sorting separately and `html` does not. Is there a reason?

Reply:
See:

https://github.com/kensanata/mastodon-backup/blob/3de4398bea58b36c8a5044ab2107629a460fb774/mastodon_archive/text.py#L64

Reply:
Yeah, that's basically "work in progress" because of #61 where I'm trying to get @frainz-de involved. :smiley:

Reply:
Ok, so I leave it up for grabs. The HTML export looks reasonable without sorting, the latest posts come on top.

Reply:
Works for me!